### PR TITLE
use extrapolator interface

### PR DESCRIPTION
**Pull request by schwoere**
Signed-off-by: mschworer <mschworer@lyft.com>

This integrates the extrapolator interface into the 3D trajectory builder. The construction is now done within the interface and local trajectory builder 3D just makes use of the interface. No functional changes.

### CODE REVIEW

File: cartographer/mapping/pose_extrapolator_interface.cc [hunk 1]
```diff
@@ -51,6 +51,7 @@ PoseExtrapolatorInterface::CreateWithImuData(
     const proto::PoseExtrapolatorOptions& options,
     const std::vector<sensor::ImuData>& imu_data) {
   CHECK(!imu_data.empty());
+  CHECK(!options.use_imu_based()) << "Not implemented!";
```
Comment (wohe) [inline cartographer/mapping/pose_extrapolator_interface.cc:54]: Add a `TODO` that the intent is to implement this?

File: cartographer/mapping/internal/3d/local_trajectory_builder_options_3d.cc [hunk 2]
```diff
@@ -57,8 +58,10 @@ proto::LocalTrajectoryBuilderOptions3D CreateLocalTrajectoryBuilderOptions3D(
           parameter_dictionary->GetDictionary("ceres_scan_matcher").get());
   *options.mutable_motion_filter_options() = CreateMotionFilterOptions(
       parameter_dictionary->GetDictionary("motion_filter").get());
+  *options.mutable_pose_extrapolator_options() = CreatePoseExtrapolatorOptions(
+     parameter_dictionary->GetDictionary("pose_extrapolator").get());
```
Comment (wohe) [inline cartographer/mapping/internal/3d/local_trajectory_builder_options_3d.cc:62]: Did you run clang-format? I think this needs to be indented by 4.

File: cartographer/mapping/internal/2d/local_trajectory_builder_options_2d.cc [hunk 3]
```diff
@@ -58,8 +59,10 @@ proto::LocalTrajectoryBuilderOptions2D CreateLocalTrajectoryBuilderOptions2D(
           parameter_dictionary->GetDictionary("ceres_scan_matcher").get());
   *options.mutable_motion_filter_options() = mapping::CreateMotionFilterOptions(
       parameter_dictionary->GetDictionary("motion_filter").get());
+  *options.mutable_pose_extrapolator_options() = CreatePoseExtrapolatorOptions(
+      parameter_dictionary->GetDictionary("pose_extrapolator").get());
   options.set_imu_gravity_time_constant(
-      parameter_dictionary->GetDouble("imu_gravity_time_constant"));
+    parameter_dictionary->GetDouble("imu_gravity_time_constant"));
```
Comment (wohe) [inline cartographer/mapping/internal/2d/local_trajectory_builder_options_2d.cc:65]: Same here: clang-format?

File: cartographer/mapping/pose_extrapolator_interface.cc (hunk 1 above)
Comment (schwoere) [reply cartographer/mapping/pose_extrapolator_interface.cc:55]: done

File: cartographer/mapping/internal/3d/local_trajectory_builder_options_3d.cc (hunk 2 above)
Comment (schwoere) [reply cartographer/mapping/internal/3d/local_trajectory_builder_options_3d.cc]: Good catch! Done.

File: cartographer/mapping/internal/2d/local_trajectory_builder_options_2d.cc (hunk 3 above)
Comment (schwoere) [reply cartographer/mapping/internal/2d/local_trajectory_builder_options_2d.cc]: done

**Pull request merged**